lion/io/database_parameters.h

```diff
@@ -50,39 +50,51 @@ inline void read_parameters(Xml_document& doc, const std::string& path, const st
 {
     assert(p.size() == used_parameters.size());
 
+    Xml_element root = doc.get_root_element();
+    Xml_element element = doc.get_element(path.substr(0, path.find_last_of("/")));
+
+    while (element.get_name() != root.get_name() && \
```

Comment:
This wouldn't work for the following XML file, no?

```xml
<a>
   <a>
        <c/>
   </a>
</a>
```

The parent must be identified in a different way than by name (see what is the way Tinyxml2 recommends to identify the parent).

Reply:
Also, why the backslash in `element.get_name() != root.get_name() && \` ?

Reply:
Sorry I missed this case. What do you think about checking until the parent of a node is returned as null?

Reply:
Yeah, this is perfect!